### DISCUSSION TITLE
Logs fom com.typesafe.scalalogging.Logger aren't printed

### DISCUSSION THREAD

**Discussion by hjenryin** [scalameta/metals#4287] [Q&A] [answered]
I was using the _METALS_  extension on vscode for a project. My _sbt_ version is 1.6.2 and _scala_ version is 2.12.8, which is required and shouldn't be changed. 
In my project I use ```com.typesafe.scalalogging.Logger``` to create logs: ```Logger("Log").info("Details")```. Also I use ```logback.xml``` to format the output:
```
<appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
        <encoder class="ch.qos.logback.classic.encoder.PatternLayoutEncoder">
            <pattern>%d{yyyy-MM-dd HH:mm:ss} [%thread] %-5level %logger{50} - %msg%n</pattern>
        </encoder>
    </appender>
```

However, after I run my code by ```run | debug```, only ```println``` goes to the _debug console_. I can't find my log.
This shouldn't be the case since the very same code works perfectly fine in _Intellij Idea_: the log is printed.

Is there a way to settle this issue? I've tried adding ```-Dmetals.loglevel=info``` to the server properties, though I think this is another level of logging and has nothing to do with ```com.typesafe.scalalogging.Logger``` (and this doesn't work of course). I'm also considering printing the output in _terminal_ in stead of _debug console_, since in ```logback.xml``` it wrote ```appender name="STDOUT"```, but I have no idea how to do that or whether it could work.

Any help or insight is appreciated! :-)

**Comment by tgodzik** [comment, marked as answer]
Thanks for the question! I think the issue here might be that the `logback.xml` doesn't end up on the proper classpath :thinking: 

This might be related to https://github.com/scalacenter/bloop/issues/1631 or https://github.com/scalacenter/bloop/issues/1504

Does it look like any of the above might bre related to your particular scenario?

You might use sbt BSP server as an alternative using `Metals: Switch Build server` command

**Reply by hjenryin** [reply]
Thank you for the suggestion! It does have something to do with the classpath: it contains multiple logging bindings, and vscode and idea happened to pick different one. Removing one will fix the problem.
